packages/vm/src/parsed_wasm.rs

```diff
@@ -73,6 +85,22 @@ impl<'a> ParsedWasm<'a> {
                         }
                     }
                 }
+                Payload::FunctionSection(section) => {
+                    // in valid wasm, the function section always has to come after the type section
+                    // (see https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#modules%E2%91%A0%E2%93%AA),
+                    // so we can assume that the type_params map is already filled at this point
+
+                    for a in section {
+                        let type_index = a? as usize;
```

Comment:
Can we keep this u32? Seems like the correct size. We only get usize above because of `.enumerate()` but could write the loop to produce u32 indices right away.

Alternatively, we can probably use a vector

Reply:
Changed to a vector now, as that makes more sense with the indexing anyways